docs-crawler/docs/docs/spiders/tf_spider.py

```diff
@@ -6,46 +12,73 @@
 
 
 class TfSpider(CrawlSpider):
-    name = "tf"
-    version = "2.1"
-    allowed_domains = ['tensorflow.org']
+    name = "tf"  # Name of the web crawler
+    version = "2.1"  # Version of TesnorFlow documentation to crawl
+    allowed_domains = ['tensorflow.org']  # Crawls only links from TensorFlow
+
+    # Base URL for crawling
     start_urls = [
-        f'https://www.tensorflow.org/versions/r{version}/api_docs/python/tf'
-    ]
+        f'https://www.tensorflow.org/versions/r{version}/api_docs/python/tf']
+
+    # Regex rules for compiling a string to a Regex object.
```

Comment:
The comment here should explain what the regex rules are. Here the rules will match on two groups and expects a function call similar to below
`foo.bar(arg1, arg2)`
First group is `foo.bar`.
Second group is `arg1, arg2`